tf-dev-override.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+set -ex
+
+# required for go install terraform-provider-google-beta
+go mod download github.com/hashicorp/terraform-plugin-mux
+
+go install github.com/hashicorp/terraform-provider-google-beta
+
+# create terraform configuration file
+cat <<EOF > /tmp/tf-dev-override.tfrc
+provider_installation {
+  # Developer overrides will stop Terraform from downloading the listed
+  # providers their origin provider registries.
+  dev_overrides {
+      "hashicorp/google-beta" = "$GOPATH/bin"
+  }
+  # For all other providers, install them directly from their origin provider
+  # registries as normal. If you omit this, Terraform will _only_ use
+  # the dev_overrides block, and so no other providers will be available.
+  # Without this, show "Failed to query available provider packages"
+  # at terraform init
+  direct{}
+}
+EOF
+
+# if the test code detects this env is set, it will add into the cmd execute.
+export TF_CLI_CONFIG_FILE="/tmp/tf-dev-override.tfrc"
```

Comment:
It looks like this isn't getting passed to the integration tests - when I modify the Makefile to run the integration tests with `-v` I don't see the following in the test output (it should show up right about `Terraform has been successfully initialized!`):

```
╷
│ Warning: Provider development overrides are in effect
│ 
│ The following provider development overrides are set in the CLI configuration:
│  - hashicorp/google in /Users/stephenrlewis/go/bin
│  - hashicorp/google-beta in /Users/stephenrlewis/go/bin
│ 
│ Skip terraform init when using provider development overrides. It is not necessary and may error unexpectedly.
╵
```

I'm pretty sure that exporting a variable in a bash script doesn't set it in the containing environment - or it might be a Makefile thing. Either way - I'm pretty sure it's not working at the moment.

Reply:
Changed to adding the var at the go test cmd